sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryAsyncClient.java

```diff
@@ -259,6 +259,128 @@ public Mono<Response<LogsBatchQueryResultCollection>> queryBatchWithResponse(Log
+            .map(result -> LogsQueryHelper.toObject(result.getTable(), type));
+    }
+
+    /**
+     * Returns all the Azure Monitor logs matching the given query for an Azure resource.
+     *
+     * @param resourceId The resourceId where the query should be executed.
+     * @param query The Kusto query to fetch the logs.
+     * @param timeInterval The time period for which the logs should be looked up.
+     * @param type The type the result of this query should be mapped to.
+     * @param <T> The type the result of this query should be mapped to.
+     * @param options The log query options to configure server timeout, set additional workspaces or enable
+     * statistics and rendering information in response.
+     * @return The logs matching the query as a list of objects of type T.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public <T> Mono<List<T>> queryResource(String resourceId, String query, QueryTimeInterval timeInterval,
+                                            Class<T> type, LogsQueryOptions options) {
+        return queryResourceWithResponse(resourceId, query, timeInterval, options, Context.NONE)
+            .map(response -> LogsQueryHelper.toObject(response.getValue().getTable(), type));
+    }
+
+    /**
+     * Returns all the Azure Monitor logs matching the given query for an Azure resource.
+     *
+     * <p><strong>Query logs from the last 7 days and set the service timeout to 2 minutes</strong></p>
+     *
+     * <!-- src_embed com.azure.monitor.query.LogsQueryAsyncClient.queryResourceWithResponse#String-String-QueryTimeInterval-LogsQueryOptions -->
+     * <pre>
+     * Mono&lt;Response&lt;LogsQueryResult&gt;&gt; queryResult = logsQueryAsyncClient.queryResourceWithResponse&#40;&quot;&#123;resource-id&#125;&quot;,
+     *     &quot;&#123;kusto-query&#125;&quot;,
+     *     QueryTimeInterval.LAST_7_DAYS,
+     *     new LogsQueryOptions&#40;&#41;.setServerTimeout&#40;Duration.ofMinutes&#40;2&#41;&#41;&#41;;
+     *
+     * queryResult.subscribe&#40;result -&gt; &#123;
+     *     for &#40;LogsTableRow row : result.getValue&#40;&#41;.getTable&#40;&#41;.getRows&#40;&#41;&#41; &#123;
+     *         System.out.println&#40;row.getRow&#40;&#41;
+     *             .stream&#40;&#41;
+     *             .map&#40;LogsTableCell::getValueAsString&#41;
+     *             .collect&#40;Collectors.joining&#40;&quot;,&quot;&#41;&#41;&#41;;
+     *     &#125;
+     * &#125;&#41;;
+     * </pre>
+     * <!-- end com.azure.monitor.query.LogsQueryAsyncClient.queryResourceWithResponse#String-String-QueryTimeInterval-LogsQueryOptions -->
+     *
+     * @param resourceId The resourceId where the query should be executed.
+     * @param query The Kusto query to fetch the logs.
+     * @param timeInterval The time period for which the logs should be looked up.
+     * @param options The log query options to configure server timeout, set additional workspaces or enable
```

Comment:
For resource queries, are workspaces still relevant? If not, then can proabbly remove the `set additional workspaces` bit from the options doc string.

Reply:
The service accepts it and when I tested it, the additional workspaces did change the response. So, we should support setting additional workspaces.